### PR TITLE
MAT-1149	Create Mongoose Models

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by AndrewBird81** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5738534&bid=18771463)

File: src/templates/mongoose/templateHelpers.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { mongoosePrimitiveTypes } from '../../model/dataTypes/primitiveDataTypes';
+import DataType from "../../model/dataTypes/DataType";
+
+export function getMongoosePrimitive(type: DataType): string {
+  // TODO: this is a workaround for the typeName holding a normalized name instead of an original name
+  let typeName = type.typeName.replace("Primitive", "").toLowerCase();
```
Comment (AndrewBird81) [inline src/templates/mongoose/templateHelpers.ts:6]: Codacy found an issue: ['typeName' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5738534)

File: src/generators/MongooseTypeGenerator.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import Generator from "./Generator";
+import FilePath from "../model/dataTypes/FilePath";
+import EntityDefinition from "../model/dataTypes/EntityDefinition";
+
+async function generate(
+  entityDefinition: EntityDefinition,
+  baseDirectory: FilePath
+): Promise<string> {
+  // TODO: work around primitive type names issue
+  let primitiveTypeName = entityDefinition.dataType.typeName.replace("Primitive", "").toLowerCase();
```
Comment (AndrewBird81) [inline src/generators/MongooseTypeGenerator.ts:13]: Codacy found an issue: ['primitiveTypeName' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5738534)

File: src/templates/mongoose/templateHelpers.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  return mongoosePrimitiveTypes[typeName] || typeName;
+}
+
+const SCHEMA_FUNCTION_REQUIRED = new Set(['DomainResource', 'Resource', 'BackboneElement', 'Element', 'Quantity', 'ElementDefinition']);
```
Comment (AndrewBird81) [inline src/templates/mongoose/templateHelpers.ts:10]: Codacy found an issue: [Variable name SCHEMA_FUNCTION_REQUIRED must match one of the following formats: camelCase](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5738534)

File: package.json [hunk 4]
```diff
@@ -37,6 +39,7 @@
     "@types/faker": "^4.1.12",
     "@types/jest": "^25.2.3",
     "@types/lodash": "^4.14.153",
+    "@types/mongoose": "^5.7.27",
```
Comment (AndrewBird81) [inline package.json:42]: Do we actually need mongoose in this project? The templates don't really refer to it

Comment (serhii-ilin) [reply package.json:42]: Right.. We probably don't. I wanted to add a simple integration test to validate mongoose mappings

File: test/templates/helpers/templateHelpers.test.ts [hunk 5]
```diff
@@ -26,4 +27,30 @@
       expect(getMongoidPrimitive("xhtml")).toEqual("String");
     });
   });
+
+  // describe("getMongoosePrimitive()", () => {
```
Comment (AndrewBird81) [inline test/templates/helpers/templateHelpers.test.ts:31]: Codacy found an issue: [Some tests seem to be commented](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5738534)

**Comment by AndrewBird81** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5738534&bid=18771463)

File: test/templates/helpers/templateHelpers.test.ts [hunk 6]
```diff
@@ -26,4 +27,30 @@
       expect(getMongoidPrimitive("xhtml")).toEqual("String");
     });
   });
+
```
Comment (AndrewBird81) [inline test/templates/helpers/templateHelpers.test.ts:30]: Codacy found an issue: [Some tests seem to be commented](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5738534)

File: test/templates/helpers/templateHelpers.test.ts [hunk 7]
```diff
@@ -2,3 +1,5 @@
+import {getMongoidPrimitive} from "../../../src/templates/helpers/templateHelpers";
```
Comment (AndrewBird81) [inline test/templates/helpers/templateHelpers.test.ts:1]: Codacy found an issue: ['getMongoosePrimitive' is defined but never used.](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5738534)

**Comment by AndrewBird81** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ src/generators/MongoidTypeGenerator.ts  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5738534&bid=18771463)

**Review by adongare — APPROVED** [review]

**Pull request merged**